src/games/strategy/triplea/ai/proAI/data/ProMoveOptions.java

```diff
@@ -0,0 +1,116 @@
+package games.strategy.triplea.ai.proAI.data;
+
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ProMoveOptions {
+
+  private final ProUtils utils;
+  private final ProBattleUtils battleUtils;
+  private final Map<Territory, ProTerritory> maxMoveMap;
+  private final Map<Territory, List<ProTerritory>> moveMaps;
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils) {
+    this.utils = utils;
+    this.battleUtils = battleUtils;
+    maxMoveMap = new HashMap<Territory, ProTerritory>();
+    moveMaps = new HashMap<Territory, List<ProTerritory>>();
+  }
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils,
+      final List<Map<Territory, ProTerritory>> moveMapList, final PlayerID player, final boolean isAttacker) {
+    this(utils, battleUtils);
+    populateMaxMoveMap(moveMapList, player, isAttacker);
+    populateMoveMaps(moveMapList);
+  }
+
+  public ProTerritory getMax(final Territory t) {
+    return maxMoveMap.get(t);
+  }
+
+  public List<ProTerritory> getAll(final Territory t) {
+    return moveMaps.get(t);
+  }
+
+  @Override
+  public String toString() {
+    return maxMoveMap.toString();
+  }
+
+  private void populateMaxMoveMap(final List<Map<Territory, ProTerritory>> moveMaps, final PlayerID player,
+      final boolean isAttacker) {
+
+    // Get players in turn order
+    final List<PlayerID> players = utils.getOtherPlayersInTurnOrder(player);
```

Comment:
So you could do that but it would be rather confusing since you'd need to pass 3 'player' parameters to the method. I personally think its easier to understand by separating them this way but I'm biased since I wrote it. 

Also the player order only needs calculated once where the isPlayersTurnFirst is inside a double for loop. So its better to save the list of players in order somewhere for efficiency sake.